app/src/store/wallet.ts

```diff
@@ -177,7 +177,7 @@ export default function useWalletStore() {
     network.value = markRaw(_network);
 
     // Start polling for data
-    startPolling();
+    poll();
```

Comment:
Why `poll` instead of `startPolling`? If a user changes networks, will this remove old listeners? I think it still will, but want to confirm

Reply:
I've restored the original.

I was seeing two `poll`s happening concurrently and I was trying to cut out the second one - now that we've removed the `.on('block')`, we need to investigate if this is still happening.

It could be that its just `poll` without a wallet connected and one with the wallet - but its going to result in an extra set of network requests we don't need to be making (especially if we're going to be using the graph)!

Reply:
In `startPolling`, I think if we just import `provider` and `defaultProvider` from the wallet store and use the below snippet that should remove the duplicate polling

```typescript
defaultProvider.removeAllListeners(); // provider used when no wallet is connected
provider.value.removeAllListeners(); // user's new provider, probably don't need this line but cant hurt
```

Looking at our old implementation, I think I configured the remove listeners call wrong so it was never removed from the default provider